closure-typing/calculator.js

```diff
@@ -1,5 +1,31 @@
-function calculator() {
-  // your code here
+function calculator(str) {
+    var operator = '';
+    var hereLiesOperator;
+    var result;
+    for (var i = 0; i < str.length; i++) {
+        if (str[i] == '+' || str[i] == '-' || str[i] == '*' || str[i] == '/') {
+            operator = str[i];
+            hereLiesOperator = i;
+        }
+    }
+    var first = +str.substring(0, hereLiesOperator);
```

Comment:
Be careful, vars hoisting to the top of the function, declarations like this might have some side effects